### DISCUSSION TITLE
Beginner quuestion express vs rocket

### DISCUSSION THREAD

**Discussion by avikat** [rwf2/Rocket#2140] [Questions]
Hi, I want use rocket for web devlopment, So as we use req,res in express , what can we use in rocket for exmple

app.post('/peerTxValidationHandler', function (req, res) {

    const TxValidate = ValidateSign.validate(req.body)
    res.json(TxValidate);
});

this is a js code. I know what it is doing how can we change it to by using rocket. Please help me .

**Comment by migeruj** [comment]
Hi @avikat!

You could be a little more specific how your validation works?

In any case Rocket.rs already has request validators and rust itself you can validate whatever comes in the request with a [structure](https://rocket.rs/v0.5-rc/guide/requests/#body-data). 
If you are looking for more complex things you have the option to use [rocket ad-hoc](https://rocket.rs/v0.5-rc/guide/requests/#ad-hoc-validation) validators or [crate validators](https://docs.rs/validator/0.13.0/validator/).

Best regards

**Reply by somehowchris** [reply]
Just fyi, kinda tried to implement a validator guard for things like json but its kinda fresh https://github.com/somehowchris/rocket-validation
